src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointDensityFilter.java

```diff
@@ -0,0 +1,131 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+
+/**
+ * A class that acts as a filter for breakpoint evidence.
+ * It passes only that evidence that is part of a putative cluster.
+ */
+public final class BreakpointDensityFilter implements Iterator<BreakpointEvidence> {
```

Comment:
You have a point.  Filtering is what it does, not what it is.  On the other hand, BreakpointDensityMapPartitioner states what it is, but obscures what it does.  I think I'm gonna punt.